File: nomad-samples/js/contents-api/contents/create-content.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function createContent(AUTH_TOKEN, CONTENT_DEFINITION_ID) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/content/new?contentDefinitionId=${CONTENT_DEFINITION_ID}`, {
        method: "GET",
        headers: HEADERS
    }).catch((exception) => {
        throw exception;
    });

    if (RESPONSE.ok)
    {
        const ID_JSON = await RESPONSE.json();
        const ID = ID_JSON.contentId;
    
        return INFO
    }

    await apiExceptionHandler(RESPONSE, "Creating content failed")
}
    

File: nomad-samples/admin/py/content_api/contents/create_content.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def create_content(AUTH_TOKEN, CONTENT_DEFINITION_ID):
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")
  
    API_URL = f"{ADMIN_API_URL}/content/new?contentDefinitionId={CONTENT_DEFINITION_ID}"
  
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        RESPONSE = requests.get(API_URL, headers=HEADERS)

        if not RESPONSE.ok:
            raise Exception()
        
        ID = json.loads(RESPONSE.text)
        return ID
        
        
    except:
        api_exception_handler(RESPONSE, "Get content id failed: ")
    
